state/azure/blobstorage/metadata.yaml

```diff
@@ -0,0 +1,82 @@
+# yaml-language-server: $schema=../../../component-metadata-schema.json
+schemaVersion: v1
+type: state
+name: azure.blobstorage
+version: v1
+status: stable
+title: "Azure Blob Storage"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-azure-blobstorage/
+capabilities:
+  - crud
+  - etag
+builtinAuthenticationProfiles:
+  - name: "azuread"
+authenticationProfiles:
+  - title: "Connection string"
+    description: "Authenticate using a connection string."
+    metadata:
+      - name: connectionString
+        required: true
+        sensitive: true
+        description: "Shared access policy connection string for Blob Storage."
+        example: '"BlobEndpoint=https://storagesample.blob.core.windows.net;SharedAccessSignature={KeySig}"'
+  - title: "Account Key"
+    description: "Authenticate using a pre-shared \"account key\"."
+    metadata:
+      - name: accountKey
+        required: true
+        sensitive: true
+        description: "The key to authenticate to the Storage Account."
+        example: '"my-secret-key"'
+      # accountName would be required for this auth. method but it is obligatory for any Blog Storage.
+metadata:
+  - name: accountName
```

Comment:
I believe `accountName` should go in the "Account Key" and "azuread" auth profiles, but it's not required by "Connection string"? At least, it looks like the connstring already includes the account name

Reply:
No, it is not required for "Connection String". https://github.com/dapr/components-contrib/blob/5dad8a867c62a6100a939093619b0d5bc620120e/internal/component/azure/blobstorage/client.go#L105-L109

`accountName` is required during metadata parsing (https://github.com/dapr/components-contrib/blob/5dad8a867c62a6100a939093619b0d5bc620120e/internal/component/azure/blobstorage/metadata.go#L53-L57) and when using "Account Key" authentication (https://github.com/dapr/components-contrib/blob/5dad8a867c62a6100a939093619b0d5bc620120e/internal/component/azure/blobstorage/client.go#L111-L117) but that's about it.

Reply:
Ok. We should implement proper support for connection strings, since they contain duplicate data.

Do you mind opening an issue for this please? So we won't forget